### PR TITLE
Add sortable fields example

### PR DESCRIPTION
**Pull request by XaosSintez**
Preview: http://rebilly.github.io/RebillyAPI/preview/sort-example/

Waiting for @RomanGotsiy thoughts about this. May be we can make it better, like `allOf` for parameters

### CODE REVIEW

File: spec/paths/customers.yaml [hunk 1]
```diff
@@ -9,9 +9,17 @@ get:
...
-    - $ref: "#/parameters/collectionSort"
-    - $ref: "#/parameters/collectionFields"
     - $ref: "#/parameters/collectionExpand"
+    - $ref: "#/parameters/collectionFields"
+    - name: sort
+      in: query
+      type: string
+      description: The collection items sort field and order (prefix with "-" for descending sort).
+      enum:
+        - id
```
Comment (RomanGotsiy) [inline spec/paths/customers.yaml]: From description seems that `"-id"`, `"-email"`, etc are valid values too. So you probably have to add these to enum if this spec will be using for validation in the future.

File: spec/paths/customers.yaml [hunk 2]
```diff
@@ -9,9 +9,17 @@ get:
     - $ref: "#/parameters/collectionOffset"
     - $ref: "#/parameters/collectionFilter"
     - $ref: "#/parameters/collectionQuery"
-    - $ref: "#/parameters/collectionSort"
-    - $ref: "#/parameters/collectionFields"
     - $ref: "#/parameters/collectionExpand"
+    - $ref: "#/parameters/collectionFields"
+    - name: sort
```
Comment (slavcodev) [inline spec/paths/customers.yaml]: Wonder if we can use `allOf` here

Comment (XaosSintez) [inline spec/paths/customers.yaml]: Yes, that's why i asked Roman's help, because there's no allOf as i see for params

**Comment by RomanGotsiy** [comment]
@XaosSintez enum is a correct way to do it. I understand it requires a lot of duplication. But unfortunately, `allOf` doesn't work in parameters.
What if I add additional bundle step which will build these enums automatically?

**Comment by XaosSintez** [comment]
Yes, it will be better. Can you please show example, how it will look like

File: spec/paths/customers.yaml (hunk 1 above)
Comment (XaosSintez) [inline spec/paths/customers.yaml]: This is more for our frontend guys. Because it may be not one field, like
`?sort=-createdTime,id`

**Comment by RomanGotsiy** [comment]
@XaosSintez I'm not sure right now about the exact syntax.
I will show it tomorrow morning, OK?

File: spec/paths/customers.yaml (hunk 1 above)
Comment (Teknologica) [inline spec/paths/customers.yaml]: So multiple sorting would be supported in the future?

Comment (XaosSintez) [inline spec/paths/customers.yaml]: @Teknologica actually it is already supported with format i gave above :-)

**Comment by RomanGotsiy** [comment]
OK,

First. As I have understood this param can accept a array of comma-separated values. So the proposed description is technically not correct.

The correct looks like the following:
```yaml
type: array
collectionFormat: csv
description: The collection items sort field and order (prefix with "-" for descending sort).
items:
  type: string
  enum:
    - id
    - -id
    - email
    - -email
    - ...
```

BUT this is not currently supported by ReDoc at the moment :( But this is simple thing so I can implement this during weekends.

Next. To omit duplication I can implement a build step which will inline param and merge. The proposed syntax:

```yaml
- ....
- $ref: "#/parameters/collectionExpand"
- x-merge:
  - $ref: "#/parameters/collectionSort"
  - items:
     enum:
      - id
      - -id
      - …
```
After processing this will be 

Why to use `x-merge` and not `allOf`? Because `allOf` has not obvious semantic and may be hard to implement while just merge is pretty simple.

If you all are OK with it, I can implement this during weekends

**Comment by RomanGotsiy** [comment]
@XaosSintez I have just pushed a commit with corresponding post-processing plugin to master (planed to make this as PR but just forgot to switch branch :))

So rebase this branch on master and you'll be able to use it.

The syntax is:

```yaml
x-rebillyMerge:
  - $ref: "#/parameters/collectionSort"
  - items:
     enum:
      - aaa
      - bbb
```

**Comment by slavcodev** [comment]
Wonder, why do we need to specify attributes twice, for `asc` and `desc` order, like `id` and `-id`? I think all attributes which we accept sorting, support the both order direction.

**Comment by RomanGotsiy** [comment]
@slavcodev because without it if you use swagger-based validation/testing in the future validation for `"-id"` will fail.

Also, once ReDoc has API console it won't allow user to put values with `-`

**Comment by slavcodev** [comment]
Okay, but could your extension `x-merge` to do this job?

**Comment by RomanGotsiy** [comment]
I believe we should use another keyword for this. `x-rebillyMerge` can be used in other places too.
Something like `x-sortableEnum` or maybe you have better suggestions?

**Comment by slavcodev** [comment]
`x-sortableEnum` looks okay, may be `x-sortable-enum`. So can we enumerate only unique attribute names, and the extension which prepare `swagger.json` will duplicate with `-`?

**Comment by RomanGotsiy** [comment]
yes, sure.
I will implement this plugin tomorrow morning.

**Pull request merged**